core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataUpdateSettingsService.java

```diff
@@ -164,13 +164,16 @@ public void updateSettings(final UpdateSettingsClusterStateUpdateRequest request
         Settings.Builder settingsForOpenIndices = Settings.builder();
         final Set<String> skippedSettings = new HashSet<>();
 
-        indexScopedSettings.validate(normalizedSettings, false); // don't validate dependencies here we check it below
-        // never allow to change the number of shards
+        indexScopedSettings.validate(normalizedSettings.filter(s -> Regex.isSimpleMatchPattern(s) == false  /* don't validate wildcards */),
+            false); //don't validate dependencies here we check it below never allow to change the number of shards
         for (String key : normalizedSettings.keySet()) {
             Setting setting = indexScopedSettings.get(key);
-            assert setting != null; // we already validated the normalized settings
+            boolean isWildcard =  setting == null && Regex.isSimpleMatchPattern(key);
+            assert setting != null // we already validated the normalized settings
```

Comment:
How is it ever possible to have `isWildcard` true if `setting` must not be null here?

Reply:
there is a second line to this assert || (isWildcard && normalizedSettings.hasValue(key) == false) does this make it clear?

Reply:
Oops sorry, the inline comments confused me!